index.js

```diff
@@ -97,121 +168,195 @@ Query.scale = function(_domain) {
+    let fn = generateEvaluateConditionFn(self, ast, $global, _filters, $valueFn, 'and');
+    return Promise.map(self, fn)
+      .then(matches => {
+        let results = self.filter((d,i) => matches[i]);
+
+        if (!results) {
+          throw {message:'INVALID_FILTER',filter:ref};
+        }
 
-    return Object.setPrototypeOf(results,Object.getPrototypeOf(self));
-  };
+        return Object.setPrototypeOf(results,Object.getPrototypeOf(self));
+      });
+  });
 };
 
-Query.where_not = function(_filters, $valueFn) {
+Query.where_not = function(_filters, $valueFn, $global) {
   var self = this;
-  return function $property(ref) {
-  
-    // Provide pipe delimited filtering
-    ref = ref.split(reSplitter).sort();
-    if (ref.length > 1)
-      // Solve for the first one, and then the remainder
-      return [ref[0],function(q) {
-        return [{q:q},'q.where_not.'+ref.slice(1).join(reSplitter),Object];
-      }];
-    
-    ref = ref[0];
-
-    ref = ref.split('=');
+  return createExternal(ast => {
+    let ref = astToCluesPath(ast);
+    let fn = generateEvaluateConditionFn(self, ast, $global, _filters, $valueFn, 'or');
+    return Promise.map(self, fn)
+      .then(matches => {
+        let results = self.filter((d,i) => !matches[i]);
+
+        if (!results) {
+          throw {message:'INVALID_FILTER',filter:ref};
+        }
 
-    var results;
-    if (ref.length == 2)
-      results = self.filter(function(d) {
-        return $valueFn(d[ref[0]]) != ref[1];
+        return Object.setPrototypeOf(results,Object.getPrototypeOf(self));
       });
-    else
-      results = _filters && _filters[ref[0]] && sift({$not:_filters[ref[0]]},self);
-
-    if (!results)
-      throw {message:'INVALID_FILTER',filter:ref};
-
-    return Object.setPrototypeOf(results,Object.getPrototypeOf(self));
-  };
+  });
 };
 
 // legacy alias
 Query.pick = Query.where;
 
 Query.select = function($global) {
   var self = this;
-  return function $property(ref) {
-    ref = ref.split(reSplitter).map(toDots);
-    return Promise.map(self.slice(),function(d) {
-      return Promise.reduce(ref,function(p,field) {
-        var key;
-        field = field.split('=');
-        key = field[1] || field[0];
-        field = field[0];
-        return clues(d,field,$global)
-          .catch(noop)
-          .then(function(d) {
-            if (ref.length > 1) 
-              p[key] = d;
-            else
-              return d;
-            return p;
-          });
+  return createExternal(ast => {
+    let rawPaths = ast.piped ? ast.piped : [ast];
+    let directList = rawPaths.length === 1;
+    let paths = rawPaths.map(path => {
+      if (path.equation) {
+        directList = false;
+        let p = astToCluesPath(path.equation.left);
+        return {
+          path: p,
+          key: path.equation.right
+        };
+      }
+      let p = astToCluesPath(path);
+      return {
+        path: p,
+        key: p.replace(/[.ᐉᐅ()|Λ=]/g,'_')
```

Comment:
select used to only work for 1 level deep - so this flattens it out.... 

i think thats best?  not quite sure what makes most sense here